src/linear_algebra/finite_dimensional.lean

```diff
@@ -1759,6 +1756,50 @@ theorem subalgebra.dim_eq_one_iff {S : subalgebra F E} : module.rank F S = 1 ↔
 theorem subalgebra.finrank_eq_one_iff {S : subalgebra F E} : finrank F S = 1 ↔ S = ⊥ :=
 ⟨subalgebra.eq_bot_of_finrank_one, subalgebra.finrank_eq_one_of_eq_bot⟩
 
+lemma subalgebra.bot_eq_top_iff_dim_eq_one :
+  module.rank F E = 1 ↔ (⊥ : subalgebra F E) = ⊤ :=
+begin
+  rw [← dim_top, ← subalgebra_top_dim_eq_submodule_top_dim],
+  split,
+  { exact λ h, (subalgebra.eq_bot_of_dim_one h).symm, },
+  { intro h,
+    haveI : finite_dimensional F (⊤ : subalgebra F E),
+    { rw ← h, exact subalgebra.finite_dimensional_bot, },
+    rw_mod_cast [← finite_dimensional.finrank_eq_dim, subalgebra.finrank_eq_one_iff],
+    exact h.symm, }
+end
```

Comment:
```suggestion
lemma subalgebra.bot_eq_top_iff_dim_eq_one :
  module.rank F E = 1 ↔ (⊥ : subalgebra F E) = ⊤ :=
by rw [← dim_top, ← subalgebra_top_dim_eq_submodule_top_dim, subalgebra.dim_eq_one_iff, eq_comm]

lemma subalgebra.bot_eq_top_iff_finrank_eq_one :
  finrank F E = 1 ↔ (⊥ : subalgebra F E) = ⊤ :=
by rw [← finrank_top, ← subalgebra_top_finrank_eq_submodule_top_finrank,
       subalgebra.finrank_eq_one_iff, eq_comm]
```
I guess you can delete `subalgebra.bot_eq_top_of_finrank_eq_one` as well.

Reply:
Maybe state the lemmas as top_eq_bot to save a `eq_comm` :)

Reply:
Thanks!

> I guess you can delete `subalgebra.bot_eq_top_of_finrank_eq_one` as well.

I kept it since it is a [simp] lemma but changed to proof.